runtime/compilers/rillv1/parse_dotenv.go

```diff
@@ -2,25 +2,50 @@ package rillv1
 
 import (
 	"context"
+	"fmt"
+	"maps"
 	"os"
 
 	"github.com/joho/godotenv"
 )
 
-// parseDotEnv parses the env file present at repo root
-func (p *Parser) parseDotEnv(ctx context.Context, path string) error {
+// parseAndMergeEnvironmentVariables parses the .env file at the given path and merges it with the existing env vars
+func (p *Parser) parseAndMergeEnvironmentVariables(ctx context.Context, path string) error {
+	// Unlikely, but if the path is empty, return an error
+	if path == "" {
+		return fmt.Errorf("empty path provided for .env file")
+	}
+
+	// Read the .env file
 	data, err := p.Repo.Get(ctx, path)
 	if err != nil {
+		// If the file doesn't exist, return nil to exit early
 		if os.IsNotExist(err) {
 			return nil
 		}
 		return err
 	}
+
+	// Unmarshal the .env file
 	envMap, err := godotenv.Unmarshal(data)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to unmarshal .env file at %s: %w", path, err)
```

Comment:
No need to include the `path` in the name here. We use a special error type to attach paths to errors, and it's already handled by the caller: https://github.com/rilldata/rill/blob/a4ced7d46771430003c26ba3f3fd21812c875865/runtime/compilers/rillv1/parser.go#L583-L586